[runtime][cts] Add test waiting on a semaphore for finite time and fix Vulkan driver

sogartar opened the pull request:
Fix Vulkan semaphore wait timeout if the timeout has already
passed. Don't check for a timeout on the host side.
Let Vulkan handle the timeout.
If the timeout has passed and the semaphore value is big enough,
we want an OK status to be returned.

Disable semaphore_test for Metal.

Commit:
[runtime][cts] Add test waiting on a semaphore for finite time

antiagainst reviewed (approved):
Code LGTM. But it looks we have an issue in Vulkan: https://github.com/iree-org/iree/actions/runs/8789691379/job/24120331047?pr=17126#step:9:307

Commit:
Fix Vulkan semaphore wait timeout if the timeout has already passed

Don't check for a timeout on the host side.
Let Vulkan handle the timeout.
If the timeout has passed and the semaphore value is big enough,
we want an OK status to be returned.

antiagainst reviewed (approved):
LGTM. Could you copy our commit message for the vulkan fix into the pull request message?

Commit:
Fix Metal driver analogous to Vulkan

Commit:
Revert "Fix Metal driver analogous to Vulkan"

This reverts commit 13989df1e08b4e14f6980aab8555ce2a73b515b0.

Commit:
Disable semaphore_test for Metal
